src/components/button/Button.js

```diff
@@ -1,30 +1,160 @@
 import React, { Component } from 'react';
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
+import ProgressBar from './ProgressBar';
+import { CheckmarkIcon, CrossIcon } from '../icon';
 import styles from './Button.css';
 
+const FEEDBACK_OUTCOME_VISIBLE_DURATION = 1750;
+
 class Button extends Component {
+    wrapperRef = React.createRef();
+    enterAnimation = false;
+
+    state = {
+        feedback: null,
+        feedbackOutcome: null,
+    };
+
+    componentDidMount() {
+        this.maybeHandleFeedbackChange();
+    }
+
+    componentDidUpdate(prevProps) {
+        this.maybeHandleFeedbackChange(prevProps.feedback);
+    }
+
+    componentWillUnmount() {
+        this.clearFeedbackOutcomeTimers();
+    }
+
     render() {
         const { variant, disabled, children, fullWidth, ...rest } = this.props;
+        const { feedback, feedbackOutcome } = this.state;
+        const loading = feedback === 'loading';
+        const hasFeedback = feedback === 'loading' || feedback === 'success' || feedback === 'error';
+        const finalDisabled = disabled || loading;
         const finalClassName = classNames(
             styles.button,
             styles[variant],
             fullWidth && styles.fullWidth,
+            styles[feedbackOutcome],
+            loading && styles.loading,
+            hasFeedback && styles.feedback,
+        );
+
+        const wrapperClassName = classNames(
+            styles.buttonWrapper,
+            styles[feedbackOutcome],
+            styles[variant],
+        );
+
+        const propsFeedback = this.props.feedback;
+        const hasPropsFeedback = propsFeedback === 'loading' || propsFeedback === 'success' || propsFeedback === 'error';
```

Comment:
Same here? Since the the `feedback` prop can only be one of `['none', 'loading', 'success', 'error']`?

Reply:
Also this could remove the need of `propsFeedback`, like so: `const hasPropsFeedback = this.props.feedback !== 'none';`

Reply:
👍